### PR TITLE
Add new builtin hashing function `ripemd_160`

### PR DESCRIPTION
**Pull request by paluh**
I'm reopening this as a separate PR because of the permissions issue (I'm not able to push to the branch under `IntersectMBO/plutus` anymore.

Pre-submit checklist:
- Branch
    - [x] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [ ] Changelog fragments have been written (if appropriate)
    - [ ] Relevant tickets are mentioned in commit messages
    - [ ] Formatting, PNG optimization, etc. are updated
- PR
    - [ ] (For external contributions) Corresponding issue exists and is linked in the description
    - [ ] Targeting master unless this is a cherry-pick backport
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [x] Reviewer requested

### CODE REVIEW

**Comment by paluh** [comment]
This is preliminary work which exposes RIPEMD-160 hashing primitive as a new bulitin.

* I closely followed changes which were made during introduction of `keccak_256` and `blake2b_224` functions.
* The current version of the PR depends on a branch of `cardano-base` which exposes this primitive hence the changes in `cabal.project`
* This draft contains an artificial costing model for the new builtin - I copied `keccak_256` related values to be able to test the function.

File: plutus-core/cost-model/data/builtinCostModelA.json [hunk 1]
```diff
@@ -527,6 +527,19 @@
             "type": "constant_cost"
         }
     },
+    "ripemd_160": {
+        "cpu": {
+            "arguments": {
```
Comment (kwxm) [inline plutus-core/cost-model/data/builtinCostModelA.json:532]: You must have run the benchmarks to get these numbers.  Which machine did you run them on?

File: plutus-ledger-api/test/Spec/Versions.hs [hunk 2]
```diff
@@ -76,7 +76,7 @@ testBuiltinVersions = testGroup "builtins"
          assertBool "not in l2,Vasil" $ isRight $ V2.deserialiseScript vasilPV serialiseDataExScript
          assertBool "not in l3,future" $ isRight $ V3.deserialiseScript futurePV serialiseDataExScript
     , testCase "bls,keccak256,blake2b224 only available in l3,Future and after" $
-         for_ [blsExScript, keccak256ExScript, blake2b224ExScript] $ \script -> do
+         for_ [blsExScript, keccak256ExScript, blake2b224ExScript, ripemd160ExScript] $ \script -> do
```
Comment (kwxm) [inline plutus-ledger-api/test/Spec/Versions.hs:79]: I'm not totally sure if `ripemd160` should be in here, but these tests need to be reworked anyway.

**Review by kwxm — COMMENTED** [review]
I think this all looks OK.  I'm impressed that you did the Agda as well!

However, we're hoping to merge the [costing code](https://github.com/IntersectMBO/plutus/pull/6301) for the extra bitwise builtins (which have already been merged) shortly.  Can you wait until that's done so that all of the bitwise operations are in the same place, and then rebase and put `ripemd160` after those?   This may require some rearrangements in your code: apologies for that.

File: plutus-core/cost-model/data/builtinCostModelA.json (hunk 1 above)
Comment (paluh) [reply plutus-core/cost-model/data/builtinCostModelA.json:532]: @kwxm - unfortunately I did not benchmark this as I stated above:

> This draft contains an artificial costing model for the new builtin - I copied keccak_256 related values for now.

That is for sure missing part :-( Sorry for the confusion.

File: plutus-ledger-api/test/Spec/Versions.hs (hunk 2 above)
Comment (paluh) [reply plutus-ledger-api/test/Spec/Versions.hs:79]: I can drop it if you wish :-)

**Comment by paluh** [comment]
> Can you wait until that's done so that all of the bitwise operations are in the same place, and then rebase and put ripemd160 after those? This may require some rearrangements in your code (and also quite a bit of other stuff has been merged since you opened this PR): apologies for that.

Sure. I will rebase and then we can sync how to finalize this PR. I'm really kin to have it merged - it will allow some fancy stuff on Cardano like checking Bitcoin witness under Bitcoin Tx etc.

**Comment by kwxm** [comment]
I'm going to close this in favour of #6378, which I'm just about to merge.

**Pull request closed without merging**